### DISCUSSION TITLE
Is there a way to programmatically order columns, including display cols?

### DISCUSSION THREAD

**Discussion by bucko13** [KevinVandy/material-react-table#20] [Q&A] [answered]
The very specific thing I'm trying to do is simply put the expanding row icon at the end of the row rather than the beginning. I noticed that [the table root sets](https://github.com/KevinVandy/material-react-table/blob/71bf61b5f67875809107ea7eb7b5ab9fa386ea88/src/table/MRT_TableRoot.tsx#L185-L189) the `columnDefs` by putting the displayColumns first and then the columns passed in through props. 

I see that there's a [`columnOrder` state](https://github.com/KevinVandy/material-react-table/blob/71bf61b5f67875809107ea7eb7b5ab9fa386ea88/src/table/MRT_TableRoot.tsx#L230) that could probably be overwritten however I don't see any mechanism by which to hook into the current column order, find the id I care about (`mrt-expand`) and move it to the end of the array where I want it. 

I haven't used react-table outside of this context, so I'm not super familiar with it, but it feels like being able to have access to something like the `getState` might be a useful general tool. I imagine that way I could get the current column order, update it, and pass it in to the table via the state prop. 

Not sure if this is a feature request or just an ask for help, but open to suggestions!

**Comment by KevinVandy** [comment, marked as answer]
You should be able to pass the column order in the initialState, and it will be respected. I think it's called mrt-row-expand as of version 0.24.0. There will be official docs for customizing display columns soon.

Edit: There is a new `positionExpandColumn` prop where you can specify `last` to make it the last column.

**Reply by bucko13** [reply]
Got it. So id still have to manually put in all ids in the order I want them rather than just picking the one I’m interested in and placing it? This should be fine in this particular case but I could imagine situations where it gets complicated, like if I have other display columns enabled. If they are missing from the column order entirely, would they just be hidden?

**Reply by KevinVandy** [reply]
Yeah you're kinda right if you want to have a more generic solution. I might need to add a position expansion prop, just like position actions. I had not anticipated anyone wanting to have the expansion column on the right.

**Reply by bucko13** [reply]
Awesome. Well feel free to ping me if you want any feedback or have questions about the use case. The workaround of manual ordering should work for now though. Will try it out today. Greater flexibility would definitely be nice though.

**Reply by KevinVandy** [reply]
feel free to join the discord server linked in the README or docs site for easier further discussion too

**Reply by KevinVandy** [reply]
new `positionExpandColumn` prop was added in v0.26.4. 

Included in official docs too https://www.material-react-table.com/docs/guides/detail-panel#position-expand-column-last

**Reply by bucko13** [reply]
Fantastic! Thanks for the quick turnaround. I ended up just adding a whole custom expander cell and manually adding that onto the end of the columns because I also needed some special stylings. But may end up falling back onto this now it's available.
